subsys/bluetooth/host/classic/conn_br.c

```diff
@@ -170,3 +170,104 @@ void bt_br_acl_recv(struct bt_conn *conn, struct net_buf *buf, bool complete)
 
 	net_buf_unref(buf);
 }
+
+int bt_conn_br_switch_role(const struct bt_conn *conn, uint8_t role)
+{
+	struct net_buf *buf;
+	struct bt_hci_cp_switch_role *cp;
+
+	CHECKIF(conn == NULL) {
+		LOG_DBG("conn is NULL");
+		return -EINVAL;
+	}
+
+	if (!bt_conn_is_type(conn, BT_CONN_TYPE_BR)) {
+		LOG_DBG("Invalid connection type: %u for %p", conn->type, conn);
+		return -EINVAL;
+	}
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_SWITCH_ROLE, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	bt_addr_copy(&cp->bdaddr, &conn->br.dst);
+	cp->role = role;
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_SWITCH_ROLE, buf, NULL);
+}
+
+static int bt_conn_br_read_link_policy_settings(const struct bt_conn *conn,
+						uint16_t *link_policy_settings)
+{
+	int err;
+	struct net_buf *buf;
+	struct bt_hci_cp_read_link_policy_settings *cp;
+	struct bt_hci_rp_read_link_policy_settings *rp;
+	struct net_buf *rsp;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_LINK_POLICY_SETTINGS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn->handle);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_LINK_POLICY_SETTINGS, buf, &rsp);
+	if (!err) {
+		rp = (void *)rsp->data;
+		*link_policy_settings = rp->link_policy_settings;
+	}
+
+	return err;
+}
+
+static int bt_conn_br_write_link_policy_settings(const struct bt_conn *conn,
+						 uint16_t link_policy_settings)
+{
+	struct net_buf *buf;
+	struct bt_hci_cp_write_link_policy_settings *cp;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_WRITE_LINK_POLICY_SETTINGS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn->handle);
+	cp->link_policy_settings = link_policy_settings;
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_WRITE_LINK_POLICY_SETTINGS, buf, NULL);
+}
+
+int bt_conn_br_set_role_switch_enable(const struct bt_conn *conn, bool enable)
```

Comment:
Whether is it better to change function name to `bt_conn_br_role_switch`?
```suggestion
int bt_conn_br_role_switch(const struct bt_conn *conn, bool enable)
```

Reply:
It is talked here https://github.com/zephyrproject-rtos/zephyr/pull/90126#discussion_r2151499920 and it is more aligned with the other bluetooth APIs.